src/fireEvent.ts

```diff
@@ -56,6 +56,9 @@ const isEventEnabled = (
   eventName?: string
 ) => {
   if (isTextInput(element)) return element?.props.editable !== false;
+  if (isTextInput(touchResponder)) {
```

Comment:
I'm not too sure about this. It will fix this issue indeed but saying that no event should be fired if at any point we pass through a non editable TextInput is not really correct either. For instance if I did a fireEvent.press on a non editable TextInput that's wrapped by a Pressable the press would not be run while it should. This may be a lesser evil still but there may be some other problems that we haven't thought of

Reply:
In the end fireEvent is just a low level API that calls props and we try to do too much with it so it has many unexpected behavior. How I see it long term is to fix this kind of issues using the new API userEvent but it's still at its early stage. It's not ideal that we can change text of non editable TextInputs but maybe to test those cases the easiest would be to use a custom matcher instead

Reply:
Tbh if we want to call `fireEvent.press` on a non editable TextInput that's wrapped by a Pressable the press shouldn't be noticed (at least that's the behaviour that user will experience and we already have in the browser)

Reply:
@pierrezimmermannbam could you post some test cases that you suspect wouldn't work so that @TMaszko could include in this PR? With `fireEvent` and handling Responder API we kinda reverse-engineer the event system of RN, which may actually change depending on the platform (especially web), so expect edge cases that would be great to understand and document

Reply:
The following test will fail

```tsx
test('can pres pressable that wraps non editable textinput', () => {
    const mockOnPress = jest.fn();

    const { getByPlaceholderText } = render(
      <Pressable onPress={mockOnPress}>
        <TextInput placeholder="hello" editable={false} />
      </Pressable>
    );

    fireEvent.press(getByPlaceholderText('hello'));

    expect(mockOnPress).toHaveBeenCalled();
  });
```

We consider all text inputs to be touch responders, so when we do a press targeting a TextInput, it is considered to be the touch responder, even though it may be non editable. So we fix a bug but introduce another, possibly more. I'm fine with this pr being merged, it looks like a lesser evil but I feel like we're trying to extend too much the fireEvent API so it will always be flawed and it should in my opinion be considered as just an API that calls a prop (in fact I think it should be renamed to fireProp). For the long term i believe we need the userEvent API to really fix those issues and keep the fireEvent API only as a way to call custom props.

Reply:
The main issue with fireEvent is that is has almost the same behavior no matter what prop we call, so we disable all events on non editable text inputs for instance, but we should still be able to call onPressIn

```tsx
test('can trigger onPressIn on disabled textinput', () => {
    const mockOnPressIn = jest.fn();

    const { getByPlaceholderText } = render(
      <TextInput
        placeholder="hello"
        editable={false}
        onPressIn={mockOnPressIn}
      />
    );

    fireEvent(getByPlaceholderText('hello'), 'pressIn');

    expect(mockOnPressIn).toHaveBeenCalled();
  });
```

This test fails as well but it shouldn't

Reply:
I don't get why this test should pass. Isn't it a proper behaviour to block events when component is basically `disabled`?

Reply:
The editable prop controls wether the value of the input can be modified or not but it still responds to touch events when it's not editable. You can try to press a non editable text input, the onPressIn prop is called in both cases

Reply:
I've been curious to get the understanding of how the TexInput events actually work when running in real RN app. I've build a small experiments app to test it, and submitted it as PR #1391.

The relevant experiment code is here: https://github.com/callstack/react-native-testing-library/pull/1391/files#diff-800ebe5060dbf16ec52e1cf9c65e7c2fa0f0cd86af32dcfdf913e13576831a70

For `editable: false` scenario both `onPressIn` and `onPressOut` events are being triggered (at least on iOS):

```
 LOG  Event: pressIn {"changedTouches": [[Circular]], "identifier": 1, "locationX": 57.33332824707031, "locationY": 35, "pageX": 77.33332824707031, "pageY": 146, "target": 117, "timestamp": 707054389.9047917, "touches": [[Circular]]}
 LOG  Event: pressOut {"changedTouches": [[Circular]], "identifier": 1, "locationX": 57.33332824707031, "locationY": 35, "pageX": 77.33332824707031, "pageY": 146, "target": 117, "timestamp": 707054477.4037917, "touches": []}
```

I'm yet to prepare experiments involving `Pressables`.

Reply:
I've also created a second test case, when non-editable `TextInput` is wrapped in `Pressable`:

```
<Pressable
        onPress={handlePressablePress}
        onPressIn={handlePressablePressIn}
        onPressOut={handlePressablePressOut}
        style={styles.pressable}
      >
        <TextInput
          style={styles.textInput}
          value={value}
          onChangeText={handleChangeText}
          editable={false}
          onPressIn={handlePressIn}
          onPressOut={handlePressOut}
        />
      </Pressable>
```

In this cases, when I press on the `TextInput` I get following events on iOS:
```
 LOG  Event: pressIn {"changedTouches": [[Circular]], "identifier": 1, "locationX": 165.66665649414062, "locationY": 16.333328247070312, "pageX": 205.66665649414062, "pageY": 147.3333282470703, "target": 65, "timestamp": 717273074.8629583, "touches": [[Circular]]}
 LOG  Event: pressOut {"changedTouches": [[Circular]], "identifier": 1, "locationX": 165.66665649414062, "locationY": 16.333328247070312, "pageX": 205.66665649414062, "pageY": 147.3333282470703, "target": 65, "timestamp": 717273155.2179583, "touches": []}
```
So non-editable `TextInput` normally receives `pressIn` and `pressOut` events, which seems to "consume" them, and prevent the press from reaching the wrapping `Pressable`. Running the same test with editable `TextInput` similarly "consumes" the touch event.